internal/operator-controller/controllers/clusterextensionrevision_controller.go

```diff
@@ -117,91 +117,50 @@ func (c *ClusterExtensionRevisionReconciler) reconcile(ctx context.Context, rev
 
 	revision, opts, err := c.toBoxcutterRevision(ctx, rev)
 	if err != nil {
-		meta.SetStatusCondition(&rev.Status.Conditions, metav1.Condition{
-			Type:               ocv1.ClusterExtensionRevisionTypeAvailable,
-			Status:             metav1.ConditionFalse,
-			Reason:             ocv1.ClusterExtensionRevisionReasonReconcileFailure,
-			Message:            err.Error(),
-			ObservedGeneration: rev.Generation,
-		})
+		setRetryingConditions(rev, err.Error())
 		return ctrl.Result{}, fmt.Errorf("converting to boxcutter revision: %v", err)
 	}
 
 	if !rev.DeletionTimestamp.IsZero() || rev.Spec.LifecycleState == ocv1.ClusterExtensionRevisionLifecycleStateArchived {
 		return c.teardown(ctx, rev, revision)
 	}
 
+	revVersion := rev.GetAnnotations()[labels.BundleVersionKey]
 	//
 	// Reconcile
 	//
 	if err := c.ensureFinalizer(ctx, rev, clusterExtensionRevisionTeardownFinalizer); err != nil {
-		meta.SetStatusCondition(&rev.Status.Conditions, metav1.Condition{
-			Type:               ocv1.ClusterExtensionRevisionTypeAvailable,
-			Status:             metav1.ConditionFalse,
-			Reason:             ocv1.ClusterExtensionRevisionReasonReconcileFailure,
-			Message:            err.Error(),
-			ObservedGeneration: rev.Generation,
-		})
 		return ctrl.Result{}, fmt.Errorf("error ensuring teardown finalizer: %v", err)
 	}
 
 	if err := c.establishWatch(ctx, rev, revision); err != nil {
-		meta.SetStatusCondition(&rev.Status.Conditions, metav1.Condition{
-			Type:               ocv1.ClusterExtensionRevisionTypeAvailable,
-			Status:             metav1.ConditionFalse,
-			Reason:             ocv1.ClusterExtensionRevisionReasonReconcileFailure,
-			Message:            err.Error(),
-			ObservedGeneration: rev.Generation,
-		})
-		return ctrl.Result{}, fmt.Errorf("establish watch: %v", err)
+		werr := fmt.Errorf("establish watch: %v", err)
+		setRetryingConditions(rev, werr.Error())
+		return ctrl.Result{}, werr
 	}
 
 	rres, err := c.RevisionEngine.Reconcile(ctx, *revision, opts...)
+	if rres != nil {
+		// Log detailed reconcile reports only in debug mode (V(1)) to reduce verbosity.
+		l.V(1).Info("reconcile report", "report", rres.String())
+	}
```

Comment:
We should move this after we checked for the errors, because in the case of errors we should still log with `l.Error(..)`